src/main/java/org/veupathdb/service/eda/ss/model/reducer/StreamIntersectMerger.java

```diff
@@ -1,15 +1,16 @@
 package org.veupathdb.service.eda.ss.model.reducer;
 
 import java.util.*;
-import java.util.stream.Collectors;
 import java.util.stream.IntStream;
 
 
 /**
- * This class is used to take the intersection of two or more streams of sequenced subsets.
+ * This class is used to take the intersection of two or more streams of sequenced longs. The class produces a
+ * stream of output longs, each of which are present in ALL input streams. This behavior is only well defined if
+ * all input streams are sorted.
  */
 public class StreamIntersectMerger implements Iterator<Long> {
```

Comment:
Note, nearly all the changes in this class are related to documentation and readability.

The only functional change is the bug fix in the constructor that ensures empty iterators are handled.